presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -224,6 +224,8 @@
 
     private double hyperloglogStandardErrorWarningThreshold = 0.004;
 
+    private Duration queryAnalyzerTimeout = new Duration(30, SECONDS);
```

Comment:
The query analyzer is basically the whole planner and there is a 3 minute timeout on the iterative optimizer. I would say this should default to the same as the iterative optimizer timeout otherwise the cases where the iterative optimizer takes a long time will also get canceled/interrupt.